### PR TITLE
Removed GpuMap.

### PR DESCRIPTION
**Pull request by emilwestergren**
No need for it when we have a base class `Map`.

@davidpiuva peer review

### CODE REVIEW

**Comment by davidpiuva** [comment]
Sounds like a good idea. I will have a look tomorrow.

File: source/draw/Map.ooc [hunk 1]
```diff
@@ -8,9 +8,73 @@
...
+	_ownsObject: Bool
+	init: func (=key, =value, =_ownsObject)
+	free: override func {
+		if (this _ownsObject)
+			this value free()
+		super()
+	}
+}
+
+_Map: class {
```
Comment (davidpiuva) [inline source/draw/Map.ooc]: Maybe call it `PropertyBag` or `Dictionary` to avoid confusion with shaders.

File: source/draw/Map.ooc [hunk 2]
```diff
@@ -8,9 +8,73 @@
...
+	}
+	use: virtual func
+	add: func <T> (key: String, value: T) {
+		if (T inheritsFrom(Object))
+			this _bindings add(key, value as Object, false)
+		else
+			this _bindings add(key, Cell<T> new(value), true)
+	}
+	get: func (key: String) -> Object { this _bindings get(key) value }
+	apply: func (action: Func (String, Object)) { this _bindings apply(action) }
```
Comment (davidpiuva) [inline source/draw/Map.ooc]: Most of these methods are just wrappers for _Map. Can we merge the classes or is purity important when passing _Map as an argument?

File: source/draw/Map.ooc [hunk 3]
```diff
@@ -8,9 +8,73 @@
 
 use geometry
 
+_MapEntry: class {
```
Comment (davidpiuva) [inline source/draw/Map.ooc]: A bit overkill to have a new class for each entry when we already have cells around each value. Can this inherit from Cell?

**Comment by davidpiuva** [comment]
Naming could be improved. Allocation might be possible to simplify but this can be another PR if a simpler solution is found.

**Comment by emilwestergren** [comment]
Yes, all valid points but should be in a separate PR since this one is only about removing/moving code.

**Comment by marcusnaslund** [comment]
So either 1) Get on that right away, or 2) Make an issue about it

**Comment by emilwestergren** [comment]
@davidpiuva create issue if you think it's necessary, I don't mind the current solution

**Pull request merged**